ngraph/test/op_eval/floor_mod.cpp

```diff
@@ -5,57 +5,138 @@
 #include <string>
 #include <vector>
 
-#include "gtest/gtest.h"
-
 #include "ngraph/op/floor_mod.hpp"
-#include "ngraph/runtime/host_tensor.hpp"
-#include "ngraph/validation_util.hpp"
-#include "runtime/backend.hpp"
-#include "util/test_tools.hpp"
+#include "gtest/gtest.h"
+#include "ngraph/ngraph.hpp"
+#include "util/engine/interpreter_engine.hpp"
+#include "util/engine/test_engines.hpp"
+#include "util/test_case.hpp"
+#include "util/test_control.hpp"
 
 using namespace std;
 using namespace ngraph;
 
 TEST(op_eval, floor_mod)
 {
-    auto a = make_shared<op::Parameter>(element::f32, Shape{4});
-    auto b = make_shared<op::Parameter>(element::f32, Shape{4});
-    auto floor_mod = make_shared<op::v1::FloorMod>(a, b);
-    auto fun = make_shared<Function>(OutputVector{floor_mod}, ParameterVector{a, b});
-
-    std::vector<float> a_value{5.1, -5.1, 5.1, -5.1};
-    std::vector<float> b_value{3.0, 3.0, -3.0, -3.0};
-    std::vector<float> expected_result{2.1, 0.9, -0.9, -2.1};
-
-    auto result = make_shared<HostTensor>();
-    ASSERT_TRUE(fun->evaluate({result},
-                              {make_host_tensor<element::Type_t::f32>(Shape{4}, a_value),
-                               make_host_tensor<element::Type_t::f32>(Shape{4}, b_value)}));
-    EXPECT_EQ(result->get_element_type(), element::f32);
-    EXPECT_EQ(result->get_shape(), Shape{4});
-    auto result_data = read_vector<float>(result);
-    for (size_t i = 0; i < expected_result.size(); i++)
-        EXPECT_NEAR(result_data[i], expected_result[i], 0.000001);
+    Shape shape{4};
+
+    auto A = make_shared<op::Parameter>(element::f32, shape);
+    auto B = make_shared<op::Parameter>(element::f32, shape);
+    auto f = make_shared<Function>(make_shared<op::FloorMod>(A, B), ParameterVector{A, B});
+
+    std::vector<float> a{5.1, -5.1, 5.1, -5.1};
+    std::vector<float> b{3.0, 3.0, -3.0, -3.0};
+
+    auto test_case = test::TestCase<ngraph::test::INTERPRETER_Engine>(f);
```

Comment:
I would suggest to leave the form of "op_eval" tests with strict `function->evaluate()` call and add tests that use `INTERPRETER_Engine` to "backend" tests.
@iefode Can you guide us how these "op_eval" tests should differ from the "backend" tests"?

Reply:
@pszmel As there is no strict answer, I will accept any of your choice. But I stick with my suggestion, what do you think?

Reply:
I will stick to TestCase if this is not an issue:)